tests/base.py

```diff
@@ -52,11 +58,19 @@ def setUp(self):
         self.devices = []
         if len(lst) == 0:
             raise unittest.SkipTest('No IB devices found')
-        for dev in lst:
-            c = d.Context(name=dev.name.decode())
+
+        dev_name = self.config['dev']
+        if dev_name:
+            c = d.Context(name=dev_name)
             attr = c.query_device()
             attr_ex = c.query_device_ex()
             self.devices.append((c, attr, attr_ex))
+        else:
+            for dev in lst:
```

Comment:
for dev in d.get_device_list():

and fail if self.devices is empty

I don't understand why there is a list of devices. A test should only run on a single device.

Reply:
> for dev in d.get_device_list():
> 
> and fail if self.devices is empty
> 
OK, I'll fix it.

> I don't understand why there is a list of devices. A test should only run on a single device.

Well, all the test cases that inherit from PyverbsAPITestCase start with a for loop over all the available devices, looks like this is by design...